components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -13,292 +22,241 @@ import ChannelInviteModal from 'components/channel_invite_modal';
     }
 
-    favoriteChannel = () => {
-        this.props.actions.favoriteChannel(this.props.channel.id);
+    function handleFavoriteChannel() {
+        props.favoriteChannel(props.channel.id);
         trackEvent('ui', 'ui_sidebar_channel_menu_favorite');
     }
 
-    unfavoriteChannel = () => {
-        this.props.actions.unfavoriteChannel(this.props.channel.id);
+    function handleUnfavoriteChannel() {
+        props.unfavoriteChannel(props.channel.id);
         trackEvent('ui', 'ui_sidebar_channel_menu_unfavorite');
     }
 
-    unmuteChannel = () => {
-        this.props.actions.unmuteChannel(this.props.currentUserId, this.props.channel.id);
+    function handleUnmuteChannel() {
+        props.unmuteChannel(props.currentUserId, props.channel.id);
     }
 
-    muteChannel = () => {
-        this.props.actions.muteChannel(this.props.currentUserId, this.props.channel.id);
+    function handleMuteChannel() {
+        props.muteChannel(props.currentUserId, props.channel.id);
     }
 
-    copyLink = () => {
-        copyToClipboard(this.props.channelLink);
+    function handleCopyLink() {
+        copyToClipboard(props.channelLink);
     }
 
-    handleLeaveChannel = (e: React.MouseEvent<HTMLSpanElement, MouseEvent>) => {
+    function handleLeaveChannel(e: MouseEvent<HTMLSpanElement, MouseEvent>) {
         e.preventDefault();
         e.stopPropagation();
-        if (this.isLeaving || !this.props.closeHandler) {
+
+        if (isLeaving.current || !props.closeHandler) {
             return;
         }
 
-        this.isLeaving = true;
-        trackEvent('ui', 'ui_sidebar_channel_menu_leave');
+        isLeaving.current = true;
 
-        this.props.closeHandler(() => {
-            this.isLeaving = false;
+        props.closeHandler(() => {
+            isLeaving.current = false;
         });
-    }
 
-    addMembers = () => {
-        const {channel, actions} = this.props;
+        trackEvent('ui', 'ui_sidebar_channel_menu_leave');
+    }
 
-        actions.openModal({
+    function handleAddMembers() {
+        props.openModal({
             modalId: ModalIdentifiers.CHANNEL_INVITE,
             dialogType: ChannelInviteModal,
-            dialogProps: {channel},
+            dialogProps: {channel: props.channel},
         });
         trackEvent('ui', 'ui_sidebar_channel_menu_addMembers');
     }
 
-    renderDropdownItems = () => {
-        const {intl, isUnread, isFavorite, isMuted, channel} = this.props;
-
-        let markAsRead;
-        if (isUnread) {
-            markAsRead = (
-                <Menu.ItemAction
-                    id={`markAsRead-${channel.id}`}
-                    onClick={this.markAsRead}
-                    icon={<i className='icon-mark-as-unread'/>}
-                    text={intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.markAsRead', defaultMessage: 'Mark as Read'})}
-                />
-            );
-        }
+    function handleOpenDirectionChange(openUp: boolean) {
```

Comment:
I guess this adds legibility, but unless that is the objective, wrapping the function doesn't seem to add any value, apart of creating a new function on every render. If you can use the `setOpenUp` directly, even better.

Reply:
we will loose this function in the new menus.